controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -106,35 +110,63 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 		return reconcile.Result{}, err
 	}
 
-	availableServices, err := util.GetAvailableServices(ctx, r.Client, storageCluster)
+	consumerConfigMapName := fmt.Sprintf("storageconsumer-%v", util.FnvHash(storageConsumer.Name))
+
+	resourceMap, err := r.createConsumerConfigMap(ctx, storageCluster, storageConsumer, consumerConfigMapName)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get available services configured in StorageCluster: %v", err)
+		return reconcile.Result{}, err
 	}
 
-	consumerConfigMap := &corev1.ConfigMap{}
-	consumerConfigMap.Name = fmt.Sprintf("storageconsumer-%v", util.FnvHash(storageConsumer.Name))
-	consumerConfigMap.Namespace = storageConsumer.Namespace
+	if err := r.removeSvgControllerOwner(ctx, resourceMap.GetSubVolumeGroupName(), storageConsumer.Namespace); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.updateStorageConsumer(ctx, storageCluster, storageConsumer, consumerConfigMapName); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *StorageConsumerUpgradeReconciler) createConsumerConfigMap(
+	ctx context.Context,
+	storageCluster *ocsv1.StorageCluster,
+	storageConsumer *ocsv1alpha1.StorageConsumer,
+	configMapName string,
+) (util.StorageConsumerResources, error) {
```

Comment:
The wrapping scope is this function, this mean you should not be returning this type